src/engine.ts

```diff
@@ -223,12 +238,49 @@ export default class Engine {
     //TODO manage gaia phase actions for specific factions
   }
 
+  leechingPhase(player: PlayerEnum, location: CubeCoordinates){
+    // exclude setiup rounds
+    if (this.round <=0) {
+      return;
+    } 
+    // all players excluded leecher
+    this.roundSubCommands = [];
+
+    for (const pl of this.players){     
+      if ( pl !== this.player(player)){
+        let leech = 0;
+        for (const loc of pl.data.occupied) {
+          if (this.map.distance(loc, location) < ISOLATED_DISTANCE) {
+            leech = Math.max(leech, pl.buildingValue( this.map.grid.get(loc.q, loc.r).data.building, this.map.grid.get(loc.q, loc.r).data.planet))
+          }
+        }
+        leech =  Math.min( leech,  pl.maxLeech(leech));
+        if (leech > 0) {
+          this.turnOrder.splice( this.currentPlayerTurnOrderPos +1, 0, this.players.indexOf(pl) )
+          this.roundSubCommands.push( {
+              name: Command.Leech,
+              player: this.players.indexOf(pl),
+              data: { leech }
+            }
+          )
+        }
+        }
+      }
+  }
+
   /** Next player to make a move, after current player makes their move */
   moveToNextPlayer(command : Command): PlayerEnum {
-    if ( command === Command.Pass || this.round === Round.SetupFaction || this.round === Round.SetupBuilding || this.round === Round.SetupRoundBooster) {
+    if ( command === Command.Pass || 
```

Comment:
`if (this.round <= 0 || command == Command.Pass || isSubCommand(command)) {`

(`isSubCommand` is a function to be added)